File: src/api/admin/proAuthorize.ts
import { post } from '@/utils/request'

export function updateUserExpiredTimeByDay<T>(data: Admin.ProAuthorize.ProAuthorizeUpdateUserExpiredTimeByDayReq) {
  return post<T>({
    url: '/admin/proAuthorize/updateUserExpiredTimeByDay',
    data,
  })
}

export function getUserProAuthorizeList<T>(data: Common.ListRequest) {
  return post<T>({
    url: '/admin/proAuthorize/getUserProAuthorizeList',
    data,
  })
}

export function getAuthorizeHistoryRecordByUserId<T>(userId: number) {
  return post<T>({
    url: '/admin/proAuthorize/getAuthorizeHistoryRecordByUserId',
    data: { userId },
  })
}

File: service/router/admin/proAuthorize.go
package admin

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitProAuthorizeRouter(router *gin.RouterGroup) {
	api := api_v1.ApiGroupApp.ApiAdmin.ProAuthorizeApi

	r := router.Group("proAuthorize", middleware.LoginInterceptor, middleware.AdminInterceptor)
	{
		r.POST("updateUserExpiredTimeByDay", api.UpdateUserExpiredTimeByDay)
		r.POST("getUserProAuthorizeList", api.GetUserProAuthorizeList)
		r.POST("getAuthorizeHistoryRecordByUserId", api.GetAuthorizeHistoryRecordByUserId)
	}

}
